Thingamajig.ts

```diff
@@ -0,0 +1,86 @@
+type Dimensions = {
+  width: number;
+  depth: number;
+  height: number;
+};
+interface IThingamajigHandler {
+  numberHandler: (t: Thingamajig) => void;
+  stringHandler: (t: Thingamajig) => void;
+  objectHandler: (t: Thingamajig) => void;
+}
+enum ThingamajigColor {
+  Red = "RED",
+  Blue = "BLUE",
+  White = "WHITE"
+}
+export default class Thingamajig {
+  name: string;
+  protected dimensions: Dimensions;
+  private label: any;
+  // at a given timestamp calls the function with the Thingamajig
+  private giveMeAtTimestamp: number;
+  private giveMeAt: ((t: Thingamajig) => void) | null;
+  color: ThingamajigColor;
+  constructor(
+    dimension: Dimensions,
+    name: string,
+    label: number | string | object,
```

Comment:
S: See my other comment about `label`

```suggestion
    label: "number" | "string" | "object",
```